core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -768,6 +771,26 @@ public PlanOptimizers(
                             // Must run before AddExchanges and after ReplicateSemiJoinInDelete
                             // to avoid temporarily having an invalid plan
                             new DetermineSemiJoinDistributionType(costComparator, taskCountEstimator)))));
+
```

Comment:
nit: we could just skip `DetermineTableScanNodePartitioning` execution above and move it to:
```
if (!forceSingleNode) {
  ..
} else {
  // HERE
}
```

Reply:
This is no longer equivalent after applying changes from https://github.com/trinodb/trino/pull/7331/files#r603981072